src/Authenticator/LoginHandler.php

```diff
@@ -491,6 +491,13 @@ public function redirectAfterSuccessfulLogin(): HTTPResponse
         }
         $request->getSession()->clear(static::SESSION_KEY . '.mustLogin');
 
+        // Deactivate sudo mode that was activated in doLogin()
+        $service = $this->getSudoModeService();
+        // Check if the service has a deactivate method, because it is not defined on the interface
+        if (method_exists($service, 'deactivate')) {
```

Comment:
```suggestion
        if (ClassInfo::hasMethod($service, 'deactivate')) {
```
This will be a bit more comprehensive and allow for some Silverstripe CMS magic that sometimes adds methods in other ways (e.g. extensions)

Reply:
Updated